docs/content/sips/xxx-spin-plugins.md

```diff
@@ -0,0 +1,235 @@
+- Maintain the only the latest release of each major version, where the latest version of the plugin will be name `$plugin` while the older versions will use the naming scheme `$plugin@<version>`. If a requirement for a specific version of the binary arises, the user can choose to install it from a custom source (local or remote).
+- Maintain all the releases and by default install the latest version according to `SemVer` but also provide a `--version` flag to specify particular versions.  
+
+#### General naming conventions of plugins
+
+The following naming conventions are to be followed for plugins where `$plugin` is the name of the plugin. 
+- The name of the plugin must not have "spin" as a prefix.
+- `$plugin` cannot be equal to one of the predefined `types` of plugins.
+- The plugins manifest must be named `$plugin.json`
+- The name field in the plugin manifest must be `$plugin`.
+- The binary of the plugin must be named `spin-$plugin` to distinguish it from other binaries.
+- The latest major release binaries must not contain any version numbers.
+- For older major releases, the name of the plugin `$plugin` must be suffixed with `@<major_version>`.
+- The license for the plugins must be named `$plugin.license`
+
+#### Additional naming conventions for Spin Dependant Plugins
+
+Spin Dependant Plugins can be distinguished from standalone binary plugins based on the name of the plugin. The Spin Dependant Plugins must follow the additional naming constraints.
+
+- The name of the plugin must be `spin-$type-$plugin` where `$type` is one of pre-defined types of plugins.
+
+Spin Dependant Plugins will be categorized based on the setup/functionality sharing required from the Spin binary. The plugins will be grouped together based on `type`. Some of the possible types are 
+
+- Host components
+- Loaders
+- Config providers
+- Deploy
+- Trigger
+
+## Packaging of plugins
+
+The plugin must be packaged as `$plugin.tar.gz`. There must be no directories in the archive. The archive must contain the plugin binary according to the naming conventions along with a license and optionally a readme.
+
+An example of a packaging a plugin named `test` would be as follows.
+
+```
+tar -czvf $plugin.tar.gz $plugin $plugin.license
+```
+
+which will produce `test.tar.gz`.
+
+## Implementation of the execution `spin $plugin`
+
+### Standalone Binary Plugins
+
+The Spin binary will just directly call the external binary while passing all the arguments passed to it along with a set of environment variables providing information about Spin.
+
+### Spin Dependant Plugins
+
+For each `<type>`, a new Spin execution path will be created sharing the required features/functionality before calling the Spin plugin. 
+
+So once the Spin plugin is invoked using `spin $plugin`, the Spin binary does all the functions defined for that specific `<type>` before calling `spin-$plugin` with all the required environment variables set.
```

Comment:
It's not quite clear to me what "the Spin binary does all the functions defined for that specific `<type>`" means here. I am correct to read this as: Spin will initialize a new execution environment implied by the plugin type and execute the plugin binary by invoking `spin $plugin`?

Reply:
Yes, That is correct.